sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -335,6 +335,11 @@ AZ_INLINE az_result _az_iot_provisioning_client_payload_registration_state_parse
     }
   }
 
+  if (found_assigned_hub != found_device_id)
```

Comment:
```suggestion
  if (!(found_assigned_hub && found_device_id))
```
having not equals makes me believe `false == false` would pass here. I have to look back up at the implementation to see if that is caught or not.

Reply:
or even have `if (found_assigned && found_device_id)` and then after default to `AZ_ERROR_ITEM_NOT_FOUND`

Reply:
> having not equals makes me believe false == false would pass here

And you would be correct :). C doesn't have XOR. false, false and true,true are the only acceptable values: IoT Hub either sends the hub and device or _none at all_ and instead sends the error code.